File: src/main/java/it/carloni/luca/aurora/option/Branch.java
package it.carloni.luca.aurora.option;


public enum Branch {

    INITIAL_LOAD("INITIAL_LOAD"),
    RE_LOAD("RE_LOAD"),
    SOURCE_LOAD("SOURCE_LOAD");

    private final String name;

    public String getName() { return name; }

    Branch(String name) { this.name = name; }
}

File: src/main/scala/it/luca/aurora/option/Branch.scala
package it.luca.aurora.option

object Branch extends Enumeration {

  protected case class Val(name: String) extends super.Val

  import scala.language.implicitConversions
  implicit def valueToBranchVal(x: Value): Val = x.asInstanceOf[Val]

  val InitialLoad: Val = Val("INITIAL_LOAD")
  val Reload: Val = Val("RELOAD")
  val SourceLoad: Val = Val("SOURCE_LOAD")

}
